components/Python/Scikit-learn/loan_demo/loan-analysis-comp/loan-analysis-final.py

```diff
@@ -0,0 +1,493 @@
+import numpy as np
+import pandas as pd
+
+import argparse
+import subprocess
+import sys
+
+from parallelm.mlops import StatCategory as st
+from parallelm.mlops import mlops as mlops
+from parallelm.mlops.stats.bar_graph import BarGraph
+from parallelm.mlops.stats.table import Table
+from parallelm.mlops.stats.graph import MultiGraph
+from scipy.stats import ks_2samp
+from sklearn.metrics import accuracy_score, roc_auc_score, roc_curve, confusion_matrix, classification_report
+from sklearn.model_selection import train_test_split
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import OneHotEncoder, MinMaxScaler
+
+def install(package):
```

Comment:
these installs should be done through deps field in the component.json?

Reply:
and im not sure this will work - before the install code is compiled, we have already included sklearn in the lines above it - so we should rewrite this part (if we cant install through component.json)

Reply:
We don't install sklearn so it works.
But maybe installs should be part of EE and not components?

Reply:
Each component may come with its own dependencies and therefore it does make sense to declare them at the component's level.
In any case, as Sriram commented, dependencies should be declared in the component.json, with `deps` field.